src/stores/user.store.js

```diff
@@ -65,11 +74,27 @@ class UserStore {
             this.provider.on('chainChanged', (_chainId) => window.location.reload());
             this.provider.on('accountsChanged', this.handleAccountsChanged)
         } catch (e) {
+            if(e.message === "no wallet selection"){
+                return
+            }
             console.error(e)
         }
-    };
+        finally {
+            this.connecting = false
+        }
+    }
+
+    autoConnect = async () => {
+        // read from the localstorage the wallet type
+        this.walletType = window.localStorage.getItem("walletType")
+        // try to establish a connection to the previously connected wallet
+        const newConnection = false
```

Comment:
why do you need a new variable here? and what will happen if wallet type is null?

Reply:
the new variable ls for self-documenting code using verbose names improves code readability
and maintainability
I could have just passed "false" to the connect function.

for the second thing, you mentioned I fixed it